addons/account/migrations/13.0.1.1/post-migration.py

```diff
@@ -0,0 +1,779 @@
+        WHERE aatamlr.account_analytic_tag_id IS NULL
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO account_move_line_account_tax_rel (
+            account_move_line_id, account_tax_id)
+        SELECT aml.id, ailt.tax_id
+        FROM account_invoice_line_tax ailt
+        JOIN account_move_line aml
+        ON aml.old_invoice_line_id = ailt.invoice_line_id
+        LEFT JOIN account_move_line_account_tax_rel amlatr
+        ON aml.id = amlatr.account_move_line_id AND ailt.tax_id = amlatr.account_tax_id
+        WHERE amlatr.account_tax_id IS NULL
+        """,
+    )
+    # Compute balance for Draft Invoice Lines
+    draft_invoices = env['account.move'].search([('state', 'in', ('draft', 'cancel'))])
+    for invoice_line in draft_invoices.line_ids:
+        vals = invoice_line._get_fields_onchange_subtotal()
+        invoice_line.write(vals)
+    draft_invoices.with_context(check_move_validity=False)._recompute_dynamic_lines()
+
+
+def migration_voucher_moves(env):
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_move am
+        SET (message_main_attachment_id, name, date, ref, narration, type,
+        journal_id, company_id, currency_id, partner_id, amount_tax,
+        amount_total, state, invoice_payment_state, invoice_date,
+        invoice_date_due, invoice_payment_ref, create_uid, create_date,
+        write_uid, write_date) = (
+        av.message_main_attachment_id, COALESCE(av.number, am.name),
+        COALESCE(av.date, am.date), av.reference, av.narration,
+        CASE WHEN av.voucher_type = 'purchase' THEN 'in_receipt'
+        ELSE 'out_receipt' END, av.journal_id, av.company_id,
+        av.currency_id, av.partner_id, av.tax_amount, av.amount,
+        CASE WHEN av.state = 'proforma' THEN 'posted' ELSE av.state END,
+        CASE WHEN av.state = 'cancel' THEN 'not_paid' ELSE 'paid' END,
+        av.account_date, av.date_due, av.name, av.create_uid, av.create_date,
+        av.write_uid, av.write_date)
+        FROM account_voucher av
+        WHERE am.old_voucher_id = av.id AND av.state not in (
+            'draft', 'cancel', 'proforma')
+        RETURNING am.old_voucher_id, am.id
+        """,
+    )
+    ids1 = env.cr.fetchall()
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO account_move (message_main_attachment_id, name, date,
+        ref, narration, type, journal_id, company_id, currency_id, partner_id,
+        amount_tax, amount_total, state, invoice_payment_state, invoice_date,
+        invoice_date_due, invoice_payment_ref, old_voucher_id,
+        create_uid, create_date, write_uid, write_date)
+        SELECT message_main_attachment_id, COALESCE(number, '/'), date,
+        reference, narration, CASE WHEN voucher_type = 'purchase'
+        THEN 'in_receipt' ELSE 'out_receipt' END, journal_id, company_id,
+        currency_id, partner_id, tax_amount, amount, state, 'not_paid',
```

Comment:
fix for account voucher with `state == proforma`
<pre>
currency_id, partner_id, tax_amount, amount, CASE WHEN av.state = 'proforma'
THEN 'posted' ELSE av.state END, 'not_paid',
</pre>